Enable "Create" and "Windows" menus on Mac

mself opened the pull request:
- On Mac, the "Create" and "Widows" menus were not appearing
- Qt has a "menu bar merging" feature on Mac that will remove any menus with no actions
- Add a dummy action, which will be deleted the first time the menus are shown

mself commented:
Now I can finally start exploring the rest of the Tartini app's features!

quicky2000 commented:
Could you make it specific to Mac and add an add an explaination comment to be sure it will not bz removed in futur?

mself commented:
Updated...

quicky2000 commented:
> Add a dummy action, which will be deleted the first time the menus are shown

what will perform the deletion of the dummy action? In your commit I just see the code adding action to menu but nothing related to removal.

is this Mac specific behaviour documented somewhere ? I'm curious to read more about it

mself commented:
The "Create" and "Windows" menus are dynamic.  They are rebuilt each time they are shown in `MainWindow::windowMenuAboutToShow()` and `MainWindow::newViewAboutToShow()`.  The first step of that process is to call `m_window_menu->clear()`.  This is where the dummy action will be deleted.

The only documentation of this is [here](https://doc.qt.io/qt-5/qmenubar.html#qmenubar-as-a-global-menu-bar).  This causes some menu items to be renamed or moved.  For example, if you have an action called "Exit", it will be renamed to "Quit" and moved to the File menu even if it was somewhere else.  This process can result in some top-level menus becoming empty, so apparently Qt hides any top-level menus that have no items in them.

This behavior isn't documented, but I tried adding a new "Foo" top-level menu and it was not shown.  When I added a "Bar" action to the Foo menu, then it did show up.  I couldn't find any other way to work around this problem.

mself commented:
Here's someone else who had the same problem (5 years ago!): https://stackoverflow.com/questions/25261760/menubar-not-showing-for-simple-qmainwindow-code-qt-creator-mac-os

mself commented:
I updated the comments in the code to make it more clear where the dummy items will be removed.

quicky2000 commented:
According to StackOverflow there is an other solution:
> setNativeMenuBar(False)

mself commented:
Using `menuBar()->setNativeMenuBar(false);` is not a good solution.  This causes the Tartini app not to use a Mac-style menu at the top of the screen, but instead to have the menu attached to the Tartini window (like how Windows does it).  No other Mac app works this way and it would be very confusing.

I don't see anything wrong with the solution I implemented.  This is basically a bug in Qt's support for Mac.

mself commented:
OK, I found a workaround that is a bit less ugly.  I think you will like this better.

quicky2000 commented:
> Using `menuBar()->setNativeMenuBar(false);` is not a good solution. This causes the Tartini app not to use a Mac-style menu at the top of the screen, but instead to have the menu attached to the Tartini window (like how Windows does it). No other Mac app works this way and it would be very confusing.
> 
> I don't see anything wrong with the solution I implemented. This is basically a bug in Qt's support for Mac.

I was not a critic, but I see your point.
Your new solution looks fine
